src/main/java/com/google/cloud/tools/app/ProcessCaller.java

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.app;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Joiner;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.logging.Logger;
+
+/**
+ * Calls external tools like gcloud or dev_appserver. Hides system process invocation logic from
+ * {@link Action}.
+ *
+ * <p>All the logic for generating and running commands is contained in this class.
+ */
+public class ProcessCaller {
+
+  // TODO(joaomartins): Will this work in e.g., Windows?
+  private static File DEFAULT_WORKING_DIR = new File("/");
+  private static Logger LOG = Logger.getLogger(ProcessCaller.class.getName());
+  private List<String> command;
```

Comment:
If I renamed this member to that, there wouldn't be any other "command" variable anywhere, so I think this is more legible.